src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/GetAzureRmRecoveryServicesVaultBackupCredentials.cs

```diff
@@ -0,0 +1,172 @@
+            //       We found a bug in the DPM UI which is looking for a particular namespace in the input file.
+            //       The below is a hack to circumvent this issue and this would be removed once the bug can be fixed.
+            vaultCredsFileContent = vaultCredsFileContent.Replace("Microsoft.Azure.Commands.AzureBackup.Models",
+                                                                  "Microsoft.Azure.Portal.RecoveryServices.Models.Common");
+
+            // prepare for download
+            string fileName = string.Format("{0}_{1:ddd MMM dd yyyy}.VaultCredentials", displayName, DateTime.UtcNow);
+            string filePath = System.IO.Path.Combine(targetLocation, fileName);
+            WriteDebug(string.Format("Saving Vault Credentials to file : {0}", filePath));
+
+            File.WriteAllBytes(filePath, Encoding.UTF8.GetBytes(vaultCredsFileContent));
+
+            // Output filename back to user
+            WriteObject(fileName);
+        }
+
+        /// <summary>
+        /// Upload certificate
+        /// </summary>
+        /// <param name="cert">management certificate</param>
+        /// <param name="subscriptionId">subscription Id</param>
+        /// <param name="resourceName">resource name</param>
+        /// <param name="resourceType">resource type</param>
+        /// <param name="resourceGroupName">resource group name</param>
+        /// <returns>acs namespace of the uploaded cert</returns>
+        private AcsNamespace UploadCert(X509Certificate2 cert, string subscriptionId)
+        {
+            UploadCertificateResponse response = RecoveryServicesClient.UploadCertificate(cert, this.Vault);
+
+            return new AcsNamespace(response);
+        }
+
+        /// <summary>
+        /// Generates vault creds file
+        /// </summary>
+        /// <param name="cert">management certificate</param>
+        /// <param name="subscriptionId">subscription Id</param>
+        /// <param name="resourceType">resource type</param>
+        /// <param name="displayName">display name</param>
+        /// <param name="acsNamespace">acs namespace</param>
+        /// <returns>xml file in string format</returns>
+        private string GenerateVaultCreds(X509Certificate2 cert, string subscriptionId, AcsNamespace acsNamespace)
+        {
+            try
+            {
+                return GenerateVaultCredsForBackup(cert, subscriptionId, acsNamespace);
+            }
+            catch (Exception exception)
+            {
+                throw exception;
+            }
+        }
+
+        /// <summary>
+        /// Generates vault creds file content for backup Vault
+        /// </summary>
+        /// <param name="cert">management certificate</param>
+        /// <param name="subscriptionId">subscription Id</param>
+        /// <param name="resourceType">resource type</param>
+        /// <param name="displayName">display name</param>
```

Comment:
nit: please keep the required params. (cross check in other method summaries too)